### PR TITLE
Added analytics for "passphrase created", "account verified" and "external_link_clicked"

### PR DESCRIPTION
**Pull request by prerna-parashar**
What: 

Why:

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/main/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/main/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

**Review by FloydLegendary92 — APPROVED** [review]

File: satellite/analytics/service.go [hunk 1]
```diff
@@ -175,9 +198,27 @@ func (service *Service) TrackEvent(eventName string, userID uuid.UUID) {
...
 		Event:  eventName,
 	})
 }
+
+// TrackLinkEvent sends an arbitrary event and link associated with user ID to Segment.
+// It is used for tracking occurrences of client-side events.
+func (service *Service) TrackLinkEvent(eventName string, userID uuid.UUID, link string) {
+	// do not track if the event name is an invalid client-side event
+
+	props := segment.NewProperties()
```
Comment (mobyvb) [inline satellite/analytics/service.go:212]: nit - move these two lines below the `if !service.clientEvents[eventName] {` block.

File: web/satellite/src/components/header/resourcesDropdown/ResourcesDropdown.vue [hunk 2]
```diff
@@ -20,12 +23,26 @@ import { RouteConfig } from '@/router';
...
+    private readonly analytics: AnalyticsHttpApi = new AnalyticsHttpApi();
     /**
      * Indicates if current route is onboarding tour.
      */
     public get isOnboardingTour(): boolean {
         return this.$route.path.includes(RouteConfig.OnboardingTour.path);
     }
+
+    public onDocsIconClick(): void {
+        this.analytics.linkEventTriggered(AnalyticsEvent.EXTERNAL_LINK_CLICKED,"https://documentation.storj.io");
```
Comment (mobyvb) [inline web/satellite/src/components/header/resourcesDropdown/ResourcesDropdown.vue:36]: nit - add space after comma (same comment for other IconClick events)

**Review by mobyvb — COMMENTED** [review]
just a small comment, but if this passes build, I think it looks great

File: satellite/analytics/service.go [hunk 3]
```diff
@@ -15,7 +15,10 @@ const (
 	eventSignedIn             = "Signed In"
 	eventProjectCreated       = "Project Created"
 	eventAccessGrantCreated   = "Access Grant Created"
+	eventAccountVerified      = "Account Verified"
```
Comment (mobyvb) [inline satellite/analytics/service.go:18]: we should make sure to use consistency in our naming here. Either all of these should have `event` in front of them, or none of them should have `event` in front of them

File: web/satellite/src/components/header/resourcesDropdown/ResourcesDropdown.vue (hunk 2 above)
Comment (prerna-parashar) [reply web/satellite/src/components/header/resourcesDropdown/ResourcesDropdown.vue]: Done.

File: satellite/analytics/service.go (hunk 3 above)
Comment (prerna-parashar) [reply satellite/analytics/service.go:18]: Done

**Review by mobyvb — APPROVED** [review]

**Review by VitaliiShpital — APPROVED** [review]

**Pull request merged**